src/main/java/servlets/SearchServlet.java

```
@@ -38,33 +39,102 @@
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(receipts));
   }
 
-  /** Returns ImmutableList of receipts from datastore with the same categories. */
-  private ImmutableList<Receipt> getReceiptsWithMatchingCategories(String categories) {
-    // Set filter to retrieve only receipts with categories equal to categories.
-    Filter matchingCategories = new FilterPredicate("categories", FilterOperator.EQUAL, categories);
-    Query query = new Query("Receipt");
-    query.setFilter(matchingCategories);
+  /** Creates a {@link QueryInformation} based on request parameters. */
+  private QueryInformation createQueryInformation(
+      HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String timeZoneId = request.getParameter("timeZoneId");
+    String categories = request.getParameter("categories");
+    String dateRange = request.getParameter("dateRange");
+    String store = request.getParameter("store");
+    String minPrice = request.getParameter("min");
+    String maxPrice = request.getParameter("max");
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    QueryInformation queryInformation = null;
+
+    try {
+      queryInformation =
+          new QueryInformation(timeZoneId, categories, dateRange, store, minPrice, maxPrice);
+    } catch (NullPointerException exception) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      response.getWriter().println(NULL_EXCEPTION_MESSAGE);
+    } catch (NumberFormatException exception) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      response.getWriter().println(NUMBER_EXCEPTION_MESSAGE);
+    } catch (ParseException exception) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      response.getWriter().println(PARSE_EXCEPTION_MESSAGE);
+    }
+
+    return queryInformation;
```

Instead of returning null when there is an exception, I would let this method propagate the exception back up and have error handling in doGet()